lib/std/zig/Ast.zig

```diff
@@ -19,10 +19,7 @@ errors: []const Error,
 pub const TokenIndex = u32;
 pub const ByteOffset = u32;
 
-pub const TokenList = std.MultiArrayList(struct {
-    tag: Token.Tag,
-    start: ByteOffset,
-});
+pub const TokenList = std.MultiArrayList(Token);
```

Comment:
Why has this changed? Storing only the start offset of the token is a very intentional design choice.

Reply:
I found that the tokenSlice function in Ast.zig could avoid running the tokenizer if I gave also stored the end index. Would you like me to revert that change?

Reply:
Yes, please. This tradeoff was made intentionally to save memory at the potential expense of a small amount of performance. I say "potential" because tokenization is fast enough that it could be faster to re-tokenize this one then than to get more cache misses from storing this data in memory. See Andrew's talk on DOD for details.

If the performance impact is more significant than expected perhaps we can revisit this. If you'd like to quantify the performance difference this gives on ast-check, that'd be appreciated.

Reply:
Here is the benchmark on Sema.zig *with* the changes to tokenizer.zig:
![image](https://github.com/user-attachments/assets/d72ec3c2-1435-4c0d-bcaf-328170a09371)
Here is the benchmark on Sema.zig *without* the changes to tokenizer.zig:
![image](https://github.com/user-attachments/assets/cd525579-0007-485f-a293-baedc11174ba)

Reply:
I feel that based on these findings, I should close this PR if we want the memory savings.